service/grails-app/services/org/olf/rs/PatronNoticeService.groovy

```diff
@@ -0,0 +1,132 @@
+package org.olf.rs
+
+import com.k_int.okapi.OkapiClient
+import grails.core.GrailsApplication
+import grails.gorm.multitenancy.Tenants
+import groovy.json.JsonSlurper
+import org.apache.kafka.clients.consumer.KafkaConsumer
+
+import org.olf.rs.EmailService
+import org.olf.rs.EventConsumerService
+import org.olf.rs.EventPublicationService
+import org.olf.rs.NoticePolicyNotice
+
+public class PatronNoticeService {
+
+  EmailService emailService
+  EventConsumerService eventConsumerService
+  EventPublicationService eventPublicationService
+  OkapiClient okapiClient
+  GrailsApplication grailsApplication
+  private KafkaConsumer consumer = null
+
+  @javax.annotation.PostConstruct
+  public void init() {
+    // We need another consumer because EventConsumerService consumes all topics
+    // as they come in whereas the notice triggers need to queue until they can
+    // be consumed under the aegis of a timer request
+    log.debug("Configuring event consumer for patron notice service")
+    Properties props = new Properties()
```

Comment:
I was able to make use of the existing EventPublicationService readily enough but the EventConsumerService would eat up the queue of notices to process before it was time. I briefly entertained the idea of it unsubscribing from the topic and then subscribing again but that seemed more fraught that just having another consumer instance. I copied this initialisation and have yet to investigate what these properties are in any detail as yet.